homeassistant/components/stream/__init__.py

```diff
@@ -41,11 +41,26 @@
 
 
 @bind_hass
-def request_stream(hass, stream_source, *, fmt="hls", keepalive=False, options=None):
+def request_stream(
+    hass,
+    stream_source,
+    *,
+    fmt="hls",
+    keepalive=False,
+    options=None,
+    master_playlist=True,
+):
     """Set up stream with token."""
     if DOMAIN not in hass.config.components:
         raise HomeAssistantError("Stream integration is not set up.")
 
+    if not master_playlist:  # request for regular playlist, stream should already exist
```

Comment:
The stream would still only exist in this case if preload was selected.

Reply:
I was assuming it would be called with master_playlist=False soon after it was called the first time with master_playlist=True, so the stream would still be open. But you are right, that is still tightly coupling to the current frontend implementation and not the general case.